app/jobs/result_monitor_loader.rb

```diff
@@ -0,0 +1,22 @@
+# Load a result from S3 into the db
+require 'logger'
+class ResultMonitorLoader
+  @queue = :q03_pipeline_run
+  @logger = Logger.new(STDOUT)
+
+  def self.perform(pipeline_run_id, output)
+    @logger.info("Loading #{output} for pipeline run #{pipeline_run_id}")
+    pr = PipelineRun.find(pipeline_run_id)
+    return unless pr.result_status_for(output) == PipelineRun::STATUS_LOADING_QUEUED
+    begin
+      pr.update_result_status(output, PipelineRun::STATUS_LOADING)
+      pr.send(PipelineRun::LOADERS_BY_OUTPUT[output])
```

Comment:
This is fine by me but you should be aware that a technique often used for dynamic dispatch like this is virtual methods.

Reply:
thanks!